### ISSUE TITLE
Enhancement: Use glsl-spec for GLSL

### ISSUE THREAD

**Issue by cbaggers** [kiselgra/c-mera#38]
`gl.xml` defines all the cpu side functions for the OpenGL spec but non of the glsl ones. This repo https://github.com/cbaggers/glsl-spec has the glsl functions and variables as either sexpressions or json. 

It is also in quicklisp so should be very easy to consume. 

I use this as the primary source of functions/variables for my glsl compiler and it seems to work pretty well. 

Feel free to ping me if anything in the repo seems vague and needs clarification

**Comment by kiselgra** [comment]
Yeah, I realized that this is odd a few days ago, because I'd actually want to use the GL functions in non-GLSL code. I think something got confused when I did this back in the day :) My hack is based on the original gl.xml, I have not yet checked what the spec provides for GLSL. I'll have a look at your version (and how you build it from the spec), probably going to help a lot :)

**Comment by cbaggers** [comment]
> "Eventually you'll realize the truth" - sk
> "The truth?" - n
> "There is no spec" - sk
> ".." - n
> "Then you'll begin to realize it is not khronos that will suffer. Only yourself" - sk
> "<begins screaming>" - n

So... yeah. There is no spec, my first version was to dump the glsl spec pdf to text and edit the thing by hand into a useful form. Alas the pdf is missing a bunch of information like what versions a given function is supported in, so I needed a different approach.

I found http://docs.gl/ and scraped all the manpages and then mostly managed to generate a spec. However the argument types were specified using Khronos's type shorthand (gvec, GenType, etc) which is inconsistent and requires extra logic from programs trying to use it (it's essentially a notation that's easy for humans to parse).
So I expanded all the function definitions into precisely specified ones, it was partly automated but still took a few days of hand editing.

The are also a number of errors in the manpages so I've been cleaning them up where possible.

So TLDR: There is no way to generate this, it's (as far as I can tell) the only resource of this information in a machine readable form.. And it was horrible to make :D

**Comment by kiselgra** [comment]
Thanks for the info, I always thought that this would be specified somewhere and similar to gl.xml. In that case I really appreciate your work. If at some point you want to update your lib let me know, maybe we can help :)

I'm currently toying around with a new rendering framework and so I'll probably get around to testing this all out pretty soon :)

**Comment by kiselgra** [comment]
I did not yet get very far with my rendering side-project, but I guess there will be some progress early next year. However, more to the point, we just recently pushed our new C-Mera version and it is much more generous with undeclared names. Therefore we dropped all those gl declarations :)

I'll get back to you when I did some actual work involving GL with C-Mera! :)
